### PR TITLE
Cherry-pick / un-GPT ShaUtil

### PR DESCRIPTION
**Pull request by riidefi**

### CODE REVIEW

**Comment by GnomedDev** [comment]
I dislike the Write/Read names, maybe make a class with a constructor and a toString method.

File: payload/sp/ShaUtil.hh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+#include <expected>
+#include <span>
+#include <string_view>
+
+#include "sp/FixedString.hh"
+
+namespace SP {
+
+FixedString<0x28 + 1> WriteSHA1(std::span<const u8, 0x14> bytes);
```
Comment (GnomedDev) [inline payload/sp/ShaUtil.hh:13]: Using FixedString is redundant as the length is always equal to the capacity. Just use a regular array.

File: payload/sp/ShaUtil.cc [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+#include "HexParser.hh"
+
+namespace SP {
+
+FixedString<0x28 + 1> WriteSHA1(std::span<const u8, 0x14> bytes) {
+    FixedString<0x28 + 1> hex;
+
+    for (size_t i = 0; i < bytes.size(); ++i) {
+        snprintf(&hex.m_buf[i * 2], 3, "%02x", bytes[i]);
```
Comment (GnomedDev) [inline payload/sp/ShaUtil.cc:14]: Should this be split out into HexParser?

**Review by GnomedDev — CHANGES_REQUESTED** [review]

**Comment by riidefi** [comment]
I'll revert back to SHA1ToHex, SHA1FromHex

File: payload/sp/ShaUtil.cc (hunk 2 above)
Comment (riidefi) [reply payload/sp/ShaUtil.cc:14]: I think we can drop HexParser and replace with https://en.cppreference.com/w/cpp/utility/to_chars

File: payload/sp/ShaUtil.hh (hunk 1 above)
Comment (riidefi) [reply payload/sp/ShaUtil.hh:13]: That's true. But 4 bytes is probably not that much overhead; probably worth the trade-off for the cleaner interface? If we really wanted to be efficient we would be working with a templated output iterator anyway.

Comment (GnomedDev) [reply payload/sp/ShaUtil.hh]: What cleaner interface? The problem with splitting this out into it's own commit is it is void of context. In track-packs, I cannot remember a time where I needed to do anything but `.data()` on the array returned back.

**Comment by riidefi** [comment]
I think neither of us knew of such a set of function's existence.

**Comment by GnomedDev** [comment]
Oh, huh, yeah never knew that existed

**Comment by riidefi** [comment]
Closing in favor of `hydro_bin2hex` and friends

**Pull request closed without merging**